setup.py

```diff
@@ -869,6 +875,148 @@ def prefix(directory, files):
+        './common/error_private.c',
+        './compress/zstd_ldm.c',
+        './compress/zstd_lazy.c',
+        './compress/huf_compress.c',
+        './compress/zstd_opt.c',
+        './compress/zstd_double_fast.c',
+        './compress/zstd_compress.c',
+        './compress/zstd_compress_superblock.c',
+        './compress/zstd_compress_sequences.c',
+        './compress/zstd_compress_literals.c',
+        './compress/zstd_fast.c',
+        './compress/fse_compress.c',
+        './compress/zstdmt_compress.c',
+        './compress/hist.c',
+        './decompress/zstd_decompress.c',
+        './decompress/huf_decompress.c',
+        './decompress/zstd_ddict.c',
+        './decompress/zstd_decompress.c',
+        './decompress/zstd_decompress_block.c',
+        './deprecated/zbuff_common.c',
+        './deprecated/zbuff_compress.c',
+        './deprecated/zbuff_decompress.c',
+        './legacy/zstd_v05.c',
+        './legacy/zstd_v04.c',
+        './legacy/zstd_v06.c',
+        './legacy/zstd_v07.c',
+        './legacy/zstd_v03.c',
+        './legacy/zstd_v02.c',
+        './legacy/zstd_v01.c',
+        './dictBuilder/cover.c',
+        './dictBuilder/divsufsort.c',
+        './dictBuilder/zdict.c',
+        './dictBuilder/fastcover.c',
+        ]
+
+    zstd_extra_objects = []
+    zstd_define_macros = []
+    zstd_sources = []
+    zstd_depends = []
+    zstd_include_dirs = [os.path.join(sz3_dir, "tools", "zstd")]
+    sz3_zstd_sources = []
+    for src in sz3_zstd_cmake_list:
+        items = src.split('/')
+        sz3_zstd_sources.append(os.path.join(sz3_dir, "tools", "zstd", items[1], items[2]))
+        zstd_include_dirs.append(os.path.join(sz3_dir, "tools", "zstd", items[1]))
+else:
+    sz3_zstd_sources = zstd_sources
+
+#TODO: Probably OpenMP should not be the default for this filter (sz3.config)
+sz3_extra_compile_args = ['-std=c++14', '-O3', '-ffast-math', '-fopenmp']
+sz3_extra_compile_args += ['/Ox', '/fp:fast', '/openmp']
+sz3_extra_link_args = ['-fopenmp', '/openmp', "-lm"]
+
+sz3_hdf5_plugin_source = os.path.join(sz3_hdf5_dir, "src", "H5Z_SZ3.cpp")
+
+
+sz3_plugin = HDF5PluginExtension(
+    "hdf5plugin.plugins.libh5sz3",
+    sources=[sz3_hdf5_plugin_source] + sz3_sources + zstd_sources,
+    extra_objects=zstd_extra_objects,
+    depends= zstd_depends + [os.path.join(sz3_hdf5_dir, "include", "H5Z_SZ3.hpp")],
+    include_dirs=sz3_include_dirs + zstd_include_dirs + [os.path.join(sz3_hdf5_dir, "include")],
+    define_macros=zstd_define_macros,
+    extra_compile_args=sz3_extra_compile_args,
+    extra_link_args=sz3_extra_link_args,
+    cpp11_required=True,
+    )
+
+if sys.platform.startswith('darwin'):
+    # this should be taken from the output of HDF5PluginExtension
+    sz3_zstd_sources += [os.path.join('src', 'hdf5_dl.c')]
+    zstd_include_dirs += [os.path.join('src', 'hdf5', 'include'), os.path.join('src', 'hdf5', 'include', 'darwin')]
+
+sz3_lib = ("sz3", {
+    "sources": sz3_zstd_sources,
```

Comment:
This looks to compile zstd twice since it is also add to `sz3_plugin` `sources`.

Actually, this way of adding our patch gives me an idea: do the same for all filters. Instead of adding `hdf5_dl.c`, we could add a static library with it inside

Reply:
No, because when that is done, the sz3_plugin sources for zstd are empty.